[symbology] add semicircle and quarter circle to simple markers

nirvn opened the pull request:
In a recent project, I had to use semicircle and quarter circles as part of point symbols. While this can be done through SVG marker, I find it's rather useful to have at-hand semicircle and quarter circle markers through simple markers.

![newmarkers](https://cloud.githubusercontent.com/assets/1728657/14243299/e3ebfb9c-fa7e-11e5-9c4f-cf33ba30eeb1.png)

nirvn committed:
[symbology] add semicircle, third, quarter circles to simple markers

m-kuhn commented on src/core/symbology-ng/qgsmarkersymbollayerv2.cpp:
```diff
@@ -432,6 +432,18 @@ bool QgsSimpleMarkerSymbolLayerV2::preparePath( QString name )
     mPath.addEllipse( QRectF( -1, -1, 2, 2 ) ); // x,y,w,h
     return true;
   }
+  if ( name == "semi_circle" )
```
should this be an `else if`?

nirvn replied:
@m-kuhn , you're right.

nyalldawson commented:
Good idea! Do you think it's worth having symbols for the bottom half/other quarters so they can easily be set without rotation? 

MapInfo has a bunch of these "dispersed group" symbols. It might be worth copying a few of these other ones too. See http://t.co/O6FfpSyHNz

nirvn commented:
@nyalldawson I find the use of rotation to be quite straight forward, see this example (a simplification of the symbology I used in the above-referred project):
![quarter](https://cloud.githubusercontent.com/assets/1728657/14244349/cd23d686-fa84-11e5-8e27-a6c17ad17e92.png)

That one quarter_circle symbol is rotated to 90, 180, 270 degrees and works like a charm :)

nirvn commented:
@m-kuhn , I've addressed your comment and added a 1/3 circle symbol. We obviously don't want to go overboard with the circles, but I feel 1/2, 1/3, and 1/4 covers a lot of ground without being excessive.

DelazJ commented:
2.16 will be fun for symbology
Miss [feature] tag for Doc...
